addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/net/HttpRequest.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.neeo.internal.net;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation.Builder;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.jetty.http.HttpStatus;
+import org.glassfish.jersey.filter.LoggingFilter;
+import org.openhab.binding.neeo.NeeoUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class represents an HTTP session with a client
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+public class HttpRequest implements AutoCloseable {
+
+    /** the logger */
+    private final Logger logger = LoggerFactory.getLogger(HttpRequest.class);
+
+    /** The client to use */
+    private final Client client;
+
+    /**
+     * Instantiates a new request
+     */
+    public HttpRequest() {
+        client = ClientBuilder.newClient();
+
+        if (logger.isDebugEnabled()) {
```

Comment:
Yes - I was against this change because of the downside.  I would be happy to document it - but I'm not sure where you mean (in the class or in the readme.md)?